integration-tests/test_selinux.py

```diff
@@ -283,3 +283,62 @@ def check_registered():
         # Verify no SELinux denials
         denials = checker.get_denials()
         assert not denials, f"SELinux denials found:\n{denials}"
+
+
+@pytest.mark.tier2
+def test_register_unconfined_service_t_registration(insights_client):
+    """
+    :id: 38fb9519-a4cf-4677-a55a-d7f660a21a3e
+    :title: Check registration running in unconfined_service_t context
+    :description:
+        This test checks that if insights --register is invoked by other (unconfined)
+        service, the registrations is still successful and more strict confinement is
+        not applied. This is mostly needed because insights-client is used not only
+        for reporting to insights,but also to invoke other code that is shipped within
+        the insights-core python library.
+    :tags: Tier 2
+    :reference: https://issues.redhat.com/browse/CCT-1718
+    :steps:
+        1. (setup) Switch system to SELinux enforcing mode
+        2. As a root user run insights-client --register in unconfined_service_t
+           context.
+        3. Check that “insights-core” code was executed in unconfined_service_t
+           context
+    :expectedresults:
+        1. (setup) The system is running in SELinux enforcing mode
+        2. Registration is successful
+        3. The insights-client (and the children processes) did not run under different
+          (more confined) mode.
+    """
+    # Not switching explicitly to enforcing mode, the system should be already in it
+    # unless someone wanted to explicitly test with permissive mode.
```

Comment:
I see this comment to be a little misleading with the first step in docstring (line 302). What if somebody would like to run only this test? Would the test run in enforcing mode?

Reply:
Yes, I'd very much like to do that. I'll discuss it also with @zpetrace